### ISSUE TITLE
TypeError in @import/unambiguous crashes linter

### ISSUE THREAD

**Issue by Taureon** [un-ts/eslint-plugin-import-x#452]
I get this in Output everytime it's supposed to lint
```
2026-02-13 00:34:49.182 [error] An unexpected error occurred:
2026-02-13 00:34:49.182 [error] TypeError: Error while loading rule '@import/unambiguous': Cannot use 'in' operator to search for 'sourceType' in undefined
Occurred while linting /home/taureon/repositories/Wormhole/wormhole-library/types/interfaces/message.ts
    at sourceType (file:///home/taureon/repositories/Wormhole/node_modules/eslint-plugin-import-x/lib/utils/source-type.js:2:22)
    at create (file:///home/taureon/repositories/Wormhole/node_modules/eslint-plugin-import-x/lib/rules/unambiguous.js:17:13)
    at Object.create (file:///home/taureon/repositories/Wormhole/node_modules/eslint-plugin-import-x/lib/utils/create-rule.js:21:20)
    at createRuleListeners (/home/taureon/repositories/Wormhole/node_modules/eslint/lib/linter/linter.js:497:15)
    at /home/taureon/repositories/Wormhole/node_modules/eslint/lib/linter/linter.js:623:7
    at Array.forEach (<anonymous>)
    at runRules (/home/taureon/repositories/Wormhole/node_modules/eslint/lib/linter/linter.js:557:31)
    at #flatVerifyWithoutProcessors (/home/taureon/repositories/Wormhole/node_modules/eslint/lib/linter/linter.js:1248:4)
    at Linter._verifyWithFlatConfigArrayAndWithoutProcessors (/home/taureon/repositories/Wormhole/node_modules/eslint/lib/linter/linter.js:1333:43)
    at Linter._verifyWithFlatConfigArray (/home/taureon/repositories/Wormhole/node_modules/eslint/lib/linter/linter.js:1400:15)
```

my settings.json
```json
{
    "workbench.editor.wrapTabs": true,
    "editor.codeActionsOnSave": {
        "source.fixAll.ts": "always",
        "source.fixAll.eslint": "always",
        "source.removeUnusedImports": "always"
    },
    "glsllint.additionalStageAssociations": {},
    "workbench.settings.showAISearchToggle": false,
    "typescript.updateImportsOnFileMove.enabled": "always",
    "explorer.confirmDragAndDrop": false
}
```

here's my eslint.config.js file if relevant
[eslint.config.js](https://github.com/user-attachments/files/25276204/eslint.config.js)

`eslint --fix` *does* work (for some reason)

everything should be up to date, I checked the extension and any npm packages

**Comment by SukkaW** [comment]
What gets you this error output? What is supposed to lint?

Since ESLint CLI via `eslint --fix` does work, I can only assume the error is not coming from ESLint (at least not ESLint normally intended to be used).

Closing the issue for now, but if you could provide more details, I'd be more than happy to re-open the issue and investigate.

**Comment by Taureon** [comment]
> What gets you this error output? What is supposed to lint?

anything, but here is one of all the files it crashes on (which happens on [all .ts files](https://codeberg.org/Taureon/Wormhole) as far as I can tell):
(ok it doesn't wanna upload my file but it's a small one so I'll just paste its content)
```ts
// message.ts
import { Platform } from "../../platforms/platform";
import type { MessageReplyPayload } from "../messageReplyOptions";

import type { Channel } from "./channel";
import type { FileAttachment } from "./fileAttachment";
import type { Server } from "./server";
import { type User } from "./user";

export interface Message {
    platform: Platform
    id: string
    content: string
    createdAt: Date
    replyIds: string[]
    attachments: FileAttachment[]

    author: User | null
    authorId: string | null

    channel: Channel | null
    channelId: string

    server: Server | null
    serverId: string | null

    deletable: boolean
    editable: boolean

    reply(options: string | MessageReplyPayload): Promise<Message | null>
    edit(content: string): Promise<Message>
    delete(): Promise<void>
}
```

funny enough, if I comment out `@import/unambiguous`, it works:

<img width="309" height="49" alt="Image" src="https://github.com/user-attachments/assets/1ec4c7ad-c2be-486e-8789-ca87426b434b" />

And by "it works" I mean I add a redundant semicolon to the interface and it gets called out in problems and removed automatically on save.

<img width="602" height="231" alt="Image" src="https://github.com/user-attachments/assets/229098bd-cede-4122-84ef-da494d51f268" />

**Comment by SukkaW** [comment]
So lemme get this straight. The crash error came from VSCode, right? Can we obtain the ESLint version used by VSCode? Maybe the issue is actually related to the ESLint 10 breaking change?

**Comment by Taureon** [comment]
eslint --version returned v10.0.0, so what youre saying about a breaking change is most probably correct
